### PR TITLE
Sitemap and Error handling

### PR DESCRIPTION
**Pull request by ElectricMaxxx**
This PR will add the Sitemap behaviour to the website. A Sitemap link is created in the footer. and will be shown on that. A xml sitemap is also available and can be added to our google analytics acount, that we hopefully have.

Still to do:

* [x] add indents to the html sitemap based on the depths in path https://github.com/symfony-cmf/seo-bundle/issues/280
* [x] make footer nicer
* [x] check and improve error handling
* [x] wait until Seo 1.3 is released

### CODE REVIEW

**Comment by ElectricMaxxx** [comment]
@WouterJ styleCI does not like `array()` notatition. Which PHP version do we use on the website and is styleCI aware of that?

File: src/AppBundle/Document/SeoPage.php [hunk 1]
```diff
@@ -22,6 +23,18 @@ class SeoPage extends Page implements SeoAwareInterface
      */
     protected $seoMetadata;
 
+    /**
+     * @var bool
+     *
+     * @PHPCR\Boolean(property="visible_for_sitemap")
+<<<<<<< HEAD
+=======
```
Comment (ElectricMaxxx) [inline src/AppBundle/Document/SeoPage.php]: need to remove that

**Comment by ElectricMaxxx** [comment]
@dbu @wouterj  can you give me a hand to update seo-bundle to 1.3 here? Got issues with jms/serializer  while updating even with min-stability: dev.

**Comment by dbu** [comment]
why do you need the minimum-stability? for the cmf dependencies? i notice that no version numbers change in composer.json... is it possible to explicitly require the components and bundles you need with `@dev` in the version constraint? if that involves too many, you could add a line `"prefer-stable": true` which should limit what is installed from dev

**Comment by ElectricMaxxx** [comment]
@dbu whaaaa composer is cheating me. A simple `rm -rf vendor/ && composer install && composer require syfmony-cmf/seo-bunde:1.3` and no i have version 1.3.

**Comment by dbu** [comment]
you still seem to get a server error on travis-ci though

**Comment by ElectricMaxxx** [comment]
cause my push is now.

**Comment by ElectricMaxxx** [comment]
@dbu  the failing tests are fine for now, i will fix them and finish that PR here. The problem was, that i hadn't the chance to update the seo-bundle until this morning.

**Comment by ElectricMaxxx** [comment]
@dbu now i got it almost just one link is broken in the new feature and that behavior is interesting there:
On the one hand: yes, the error page looks like it should do :-) On the other hand: I created the footer menu by hand (hardcoded) in the template and have `http://127.0.0.1:8000/about/` and got:
image
Shouldn't we be a little bit more calm with that last `/`?

File: app/Resources/views/layout.html.twig [hunk 2]
```diff
@@ -54,9 +54,45 @@
...
 
     <div id="footer">
-        <a href="https://twitter.com/SymfonyCMF" class="twitter-follow-button" data-show-count="false">Follow @SymfonyCMF</a>
-        <script>!function(d,s,id){var js,fjs=d.getElementsByTagName(s)[0],p=/^http:/.test(d.location)?'http':'https';if(!d.getElementById(id)){js=d.createElement(s);js.id=id;js.src=p+'://platform.twitter.com/widgets.js';fjs.parentNode.insertBefore(js,fjs);}}(document, 'script', 'twitter-wjs');</script>
-        <br/>
+        <div class="links">
+            <div class="link-col">
+                <h2>Links</h2>
+                <ul>
+                    <li><a href="{{ url('/cms/simple/about') }}/">About</a></li>
```
Comment (ElectricMaxxx) [inline app/Resources/views/layout.html.twig]: here is the point where i did the mistake.

**Comment by wouterj** [comment]
> Shouldn't we be a little bit more calm with that last /?

It's a setting to allow/disallow this (can be set by the admin).

**Comment by ElectricMaxxx** [comment]
It is fine then. Thanks.

File: app/config/services.yml [hunk 3]
```diff
@@ -12,3 +13,25 @@ services:
             - @cmf_core.publish_workflow.checker
         tags:
             - {name: "sonata.block"}
+    cmf_simple_cms.initializer:
+        class: AppBundle\Initializer\HomePageInitializer
+        arguments:
+            - %cmf_simple_cms.persistence.phpcr.basepath%
+            - %app_bundle.document.seo_page%
```
Comment (dbu) [inline app/config/services.yml]: i think we should put the class name here directly, not define a parameter for this

File: src/AppBundle/DataFixtures/PHPCR/LoadStaticData.php [hunk 4]
```diff
@@ -92,7 +93,7 @@ public function load(ObjectManager $manager)
 
     private function loadPage(ObjectManager $manager, $basepath, $pageData)
     {
-        $class = isset($pageData['class']) ? $pageData['class'] : '\Symfony\Cmf\Bundle\SimpleCmsBundle\Doctrine\Phpcr\Page';
+        $class = isset($pageData['class']) ? $pageData['class'] : '\AppBundle\Document\SeoPage';
```
Comment (dbu) [inline src/AppBundle/DataFixtures/PHPCR/LoadStaticData.php]: could we use SeoPage::class ?

File: src/AppBundle/Document/TalkBlock.php [hunk 5]
```diff
@@ -10,7 +10,9 @@
  */
 class TalkBlock extends SimpleBlock
 {
-    /** @PHPCR\Field(type="string") */
+    /**
+     * @PHPCR\String
```
Comment (dbu) [inline src/AppBundle/Document/TalkBlock.php]: this is the old, deprecated format. we should use Field(type="string"). php 7 does not allow a class named String.

but we should make the docblock multiline like you did.

**Comment by dbu** [comment]
:confetti_ball:

**Pull request merged**